content/influxdb/cloud-iox/query-data/flux-sql.md

```diff
@@ -0,0 +1,522 @@
+---
+title: Use Flux and SQL to query data
+description: >
+  Leverage both the performance of SQL and the flexibility of Flux to query and
+  process your time series data.
+menu:
+  influxdb_cloud_iox:
+    name: Use Flux & SQL
+    parent: Query data
+weight: 204
+related:
+  - /influxdb/cloud-iox/get-started/query/
+  - /influxdb/cloud-iox/query-data/sql/
+influxdb/cloud-iox/tags: [sql, flux, query]
+---
+
+InfluxDB Cloud powered by [InfluxDB IOx](/influxdb/cloud-iox/#the-influxdb-iox-storage-engine)
+supports both [Flux](/flux/v0.x/) and [SQL](/influxdb/cloud-iox/reference/sql/) query languages.
+Flux is a full-featured data scripting language that provides a wide range of
+functionality and flexibility. SQL is a proven and performant relational query language.
+
+This guide walks through leveraging the performance of SQL and the flexibility of
+Flux when querying your time series data.
+
+{{% note %}}
+#### Sample data
+
+The query examples below use the
+[Get started sample data](/influxdb/cloud-iox/get-started/write/#write-line-protocol-to-influxdb).
+{{% /note %}}
+
+- [Performance and flexibility](#performance-and-flexibility)
+- [What to do in SQL versus Flux?](#what-to-do-in-sql-versus-flux?)
+- [Use SQL and Flux together](#use-sql-and-flux-together)
+  - [Helper functions for SQL in Flux](#helper-functions-for-sql-in-flux)
+  - [SQL results structure](#sql-results-structure)
+- [Process SQL results with Flux](#process-sql-results-with-flux)
+  - [Group by tags](#group-by-tags)
+  - [Rename the `time` column to `_time`](#rename-the-time-column-to-_time)
+  - [Unpivot your data](#unpivot-your-data)
+  - [Example SQL query with further Flux processing](#example-sql-query-with-further-flux-processing)
+
+## Performance and flexibility
+
+Flux was designed and optimized for the
+[TSM data model](/influxdb/v2.6/reference/internals/storage-engine/#time-structured-merge-tree-tsm),
+which is fundamentally different from IOx.
+Because of this, Flux is less performant when querying an IOx-powered bucket.
+However, as a full-featured scripting language, Flux gives you the flexibility
+to perform a wide range of data processing operations such as statistical
+analysis, alerting, HTTP API interactions, and other operations that aren't
+supported in SQL.
+By using Flux and SQL together, you can benefit from both the performance of SQL
+and the flexibility of Flux.
+
+## What to do in SQL versus Flux?
+
+We recommend doing as much of your query as possible in SQL for the most
+performant queries.
+Do any further processing in Flux.
+The following chain of Flux functions can be replicated in SQL:
+
+```js
```

Comment:
I tried to turn the bones of a SQL query into this format so a user can see the bare bones comparison of each before the examples but can't quite get it to work:
```sql
SELECT
  DATE_BIN() AS _time,
  avg() AS ###,
FROM <>
WHERE
  time >= ###
  AND time < ###
GROUP BY _time
ORDER BY _time
```
Your examples build the aggregate to probs isn't needed

Reply:
I want to keep the example below just to show an actual query. The examples are "collapsed" by default, so users have to opt into viewing them. I did add something similar to your proposal along with the chain of generic Flux functions.